actions/postgresql_show_create_table_action.go

```diff
@@ -0,0 +1,448 @@
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/pkg/errors"
+)
+
+type columnInfo struct {
+	Attname        string
+	FormatType     string
+	Substring      *string
+	Attnotnull     bool
+	Attcollation   *string
+	Attidentity    string
+	Attstorage     string
+	Attstattarget  *string
+	ColDescription *string
+}
+
+type indexInfo struct {
+	Relname            string
+	IsPrimary          bool
+	IsUnique           bool
+	IsClustered        bool
+	IsValid            bool
+	Indrelid           string
+	PgGetIndexDef      *string
+	PgGetConstraintDef *string
+	Contype            *string
+	Condeferrable      *bool
+	Condeferred        *bool
+	Indisreplident     *bool
+	Reltablespace      int
+}
+
+type postgresqlShowCreateTableAction struct {
+	id     string
+	params *agentpb.StartActionRequest_PostgreSQLShowCreateTableParams
+}
+
+// NewPostgreSQLShowCreateTableAction creates PostgreSQL SHOW CREATE TABLE Action.
+// This is an Action that can run `SHOW CREATE TABLE` command on PostgreSQL service with given DSN.
+func NewPostgreSQLShowCreateTableAction(id string, params *agentpb.StartActionRequest_PostgreSQLShowCreateTableParams) Action {
+	return &postgresqlShowCreateTableAction{
+		id:     id,
+		params: params,
+	}
+}
+
+// ID returns an Action ID.
+func (e *postgresqlShowCreateTableAction) ID() string {
+	return e.id
+}
+
+// Type returns an Action type.
+func (e *postgresqlShowCreateTableAction) Type() string {
+	return "postgresql-show-create-table"
+}
+
+// Run runs an Action and returns output and error.
+func (e *postgresqlShowCreateTableAction) Run(ctx context.Context) ([]byte, error) {
+	db, err := sql.Open("postgres", e.params.Dsn)
+	if err != nil {
+		return nil, err
+	}
+	defer db.Close() //nolint:errcheck
+	var buf bytes.Buffer
+	w := bufio.NewWriter(&buf)
+
+	// Extract table id
+	tableID, err := e.printTableInit(ctx, w, db)
+	if err != nil {
+		return nil, err
+	}
+
+	/* Generate table cells to be printed */
+	err = e.printColumnsInfo(ctx, w, db, tableID)
+	if err != nil {
+		return nil, err
+	}
+
+	/* print indexes */
+	err = e.printIndexInfo(ctx, w, db, tableID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = e.printCheckConstraints(ctx, w, db, tableID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = e.printForeignKeyConstraints(ctx, w, db, tableID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = e.printReferencedBy(ctx, w, db, tableID)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = w.Flush(); err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
+
+func (e *postgresqlShowCreateTableAction) printTableInit(ctx context.Context, w io.Writer, db *sql.DB) (string, error) {
+	var tableID, schema, relname string
+	row := db.QueryRowContext(ctx, `SELECT c.oid,
+	       n.nspname,
+	       c.relname
+	FROM pg_catalog.pg_class c
+	         LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
+	WHERE c.relname = $1
+	  AND pg_catalog.pg_table_is_visible(c.oid)
+	ORDER BY 2, 3;`, e.params.Table)
```

Comment:
I know that is a generated query, but let's at least replace `2, 3` with column names.